src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexFacetedReadOperation.cs

```diff
@@ -102,62 +221,63 @@ public sealed class CoraxIndexFacetedReadOperation : IndexFacetReadOperationBase
         return results.Values
             .Select(x => x.Result)
             .ToList();
+    }
 
-        void UpdateRangeResults()
+    private void UpdateRangeResults(Dictionary<string, FacetedQueryParser.FacetResult> results, Dictionary<string, Dictionary<string, FacetValues>> facetsByRange)
+    {
+        foreach (var result in results)
```

Comment:
Can we assume that results are ordered? In order to avoid the `O(n^2)` operation?

Reply:
No